pkg/archive/archive_unix.go

```diff
@@ -118,3 +119,8 @@ func handleLChmod(hdr *tar.Header, path string, hdrInfo os.FileInfo, forceMask *
 	}
 	return nil
 }
+
+// Hardlink without symlinks
+func handleLLink(targetPath, path string) error {
+	return unix.Linkat(unix.AT_FDCWD, targetPath, unix.AT_FDCWD, path, 0)
```

Comment:
Is there any reason for dropping `if err := os.Link(targetPath, path); err != nil {` for `unix` variant and directly using `Linkat(`.

Reply:
I don't think this is strictly needed. The confusing thing here when working on the tests was that the code implicitly assumed that hard links to symlinks do not follow the symlink and changing to Linkat perhaps makes that slightly clearer. It could be improved with a comment.

Looking at archive_unix.go and archive_freebsd.go, they are quite similar and it would be worth trying to share the implementation and reduce archive_freebsd.go to a minimum. Probably in a separate PR.

Reply:
I added a comment to clarify a little.